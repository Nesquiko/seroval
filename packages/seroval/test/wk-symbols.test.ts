import { describe, it, expect } from 'vitest';
import {
  // deserialize,
  // fromJSON,
  serialize,
  serializeAsync,
  toJSON,
  toJSONAsync,
} from '../src';

describe('well-known symbols', () => {
  describe('serialize', () => {
    it('supports strings', () => {
      expect(serialize(Symbol.asyncIterator)).toMatchSnapshot();
      expect(serialize(Symbol.hasInstance)).toMatchSnapshot();
      expect(serialize(Symbol.isConcatSpreadable)).toMatchSnapshot();
      expect(serialize(Symbol.iterator)).toMatchSnapshot();
      expect(serialize(Symbol.match)).toMatchSnapshot();
      expect(serialize(Symbol.matchAll)).toMatchSnapshot();
      expect(serialize(Symbol.replace)).toMatchSnapshot();
      expect(serialize(Symbol.search)).toMatchSnapshot();
      expect(serialize(Symbol.species)).toMatchSnapshot();
      expect(serialize(Symbol.split)).toMatchSnapshot();
      expect(serialize(Symbol.toPrimitive)).toMatchSnapshot();
      expect(serialize(Symbol.toStringTag)).toMatchSnapshot();
      expect(serialize(Symbol.unscopables)).toMatchSnapshot();
    });
  });
  describe('serializeAsync', () => {
    it('supports strings', async () => {
      expect(await serializeAsync(Promise.resolve(Symbol.asyncIterator))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.hasInstance))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.isConcatSpreadable))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.iterator))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.match))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.matchAll))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.replace))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.search))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.species))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.split))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.toPrimitive))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.toStringTag))).toMatchSnapshot();
      expect(await serializeAsync(Promise.resolve(Symbol.unscopables))).toMatchSnapshot();
    });
  });
  describe('toJSON', () => {
    it('supports strings', () => {
      expect(JSON.stringify(toJSON(Symbol.asyncIterator))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.hasInstance))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.isConcatSpreadable))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.iterator))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.match))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.matchAll))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.replace))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.search))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.species))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.split))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.toPrimitive))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.toStringTag))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Symbol.unscopables))).toMatchSnapshot();
    });
  });
  describe('toJSONAsync', () => {
    it('supports strings', async () => {
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.asyncIterator))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.hasInstance))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.isConcatSpreadable))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.iterator))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.match))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.matchAll))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.replace))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.search))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.species))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.split))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.toPrimitive))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.toStringTag))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Symbol.unscopables))),
      ).toMatchSnapshot();
    });
  });
});
