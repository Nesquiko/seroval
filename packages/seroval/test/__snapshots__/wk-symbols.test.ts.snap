// Vitest Snapshot v1

exports[`well-known symbols > serialize > supports strings 1`] = `"Symbol.asyncIterator"`;

exports[`well-known symbols > serialize > supports strings 2`] = `"Symbol.hasInstance"`;

exports[`well-known symbols > serialize > supports strings 3`] = `"Symbol.isConcatSpreadable"`;

exports[`well-known symbols > serialize > supports strings 4`] = `"Symbol.iterator"`;

exports[`well-known symbols > serialize > supports strings 5`] = `"Symbol.match"`;

exports[`well-known symbols > serialize > supports strings 6`] = `"Symbol.matchAll"`;

exports[`well-known symbols > serialize > supports strings 7`] = `"Symbol.replace"`;

exports[`well-known symbols > serialize > supports strings 8`] = `"Symbol.search"`;

exports[`well-known symbols > serialize > supports strings 9`] = `"Symbol.species"`;

exports[`well-known symbols > serialize > supports strings 10`] = `"Symbol.split"`;

exports[`well-known symbols > serialize > supports strings 11`] = `"Symbol.toPrimitive"`;

exports[`well-known symbols > serialize > supports strings 12`] = `"Symbol.toStringTag"`;

exports[`well-known symbols > serialize > supports strings 13`] = `"Symbol.unscopables"`;

exports[`well-known symbols > serializeAsync > supports strings 1`] = `"Promise.resolve(Symbol.asyncIterator)"`;

exports[`well-known symbols > serializeAsync > supports strings 2`] = `"Promise.resolve(Symbol.hasInstance)"`;

exports[`well-known symbols > serializeAsync > supports strings 3`] = `"Promise.resolve(Symbol.isConcatSpreadable)"`;

exports[`well-known symbols > serializeAsync > supports strings 4`] = `"Promise.resolve(Symbol.iterator)"`;

exports[`well-known symbols > serializeAsync > supports strings 5`] = `"Promise.resolve(Symbol.match)"`;

exports[`well-known symbols > serializeAsync > supports strings 6`] = `"Promise.resolve(Symbol.matchAll)"`;

exports[`well-known symbols > serializeAsync > supports strings 7`] = `"Promise.resolve(Symbol.replace)"`;

exports[`well-known symbols > serializeAsync > supports strings 8`] = `"Promise.resolve(Symbol.search)"`;

exports[`well-known symbols > serializeAsync > supports strings 9`] = `"Promise.resolve(Symbol.species)"`;

exports[`well-known symbols > serializeAsync > supports strings 10`] = `"Promise.resolve(Symbol.split)"`;

exports[`well-known symbols > serializeAsync > supports strings 11`] = `"Promise.resolve(Symbol.toPrimitive)"`;

exports[`well-known symbols > serializeAsync > supports strings 12`] = `"Promise.resolve(Symbol.toStringTag)"`;

exports[`well-known symbols > serializeAsync > supports strings 13`] = `"Promise.resolve(Symbol.unscopables)"`;

exports[`well-known symbols > toJSON > supports strings 1`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":0},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 2`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":1},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 3`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":2},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 4`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":3},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 5`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":4},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 6`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":5},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 7`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":6},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 8`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":7},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 9`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":8},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 10`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":9},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 11`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":10},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 12`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":11},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSON > supports strings 13`] = `"{\\"t\\":{\\"t\\":24,\\"s\\":12},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 1`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":0}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 2`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":1}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 3`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":2}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 4`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":3}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 5`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":4}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 6`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":5}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 7`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":6}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 8`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":7}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 9`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":8}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 10`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":9}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 11`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":10}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 12`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":11}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`well-known symbols > toJSONAsync > supports strings 13`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":24,\\"s\\":12}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;
